### PR TITLE
fix: modify reps and calories to allow only value from 1-1000

### LINKED ISSUE #160: edit addex editex r/0, c/0, r/MAX_INT, c/MAX_INT, r/000001
**Current Behaviour:**
- r/0 and c/0 throw error kill program
- r/000001 -> still displayed as 00001

**Expected Behaviour:**
- ui should only allow 1-1000 number for calories and rep
- preceding 0 should be removed from number

## Issues

[[PE-D] edit 1 r/0 does not change reps to 0 under exercise log or give any error messages · Issue #126 · AY2021S1-CS2103-F10-3/tp](https://github.com/AY2021S1-CS2103-F10-3/tp/issues/126)

[https://github.com/AY2021S1-CS2103-F10-3/tp/issues/148](https://github.com/AY2021S1-CS2103-F10-3/tp/issues/148)

[https://github.com/AY2021S1-CS2103-F10-3/tp/issues/144](https://github.com/AY2021S1-CS2103-F10-3/tp/issues/144)

[[PE-D] Edit Command Number format · Issue #131 · AY2021S1-CS2103-F10-3/tp](https://github.com/AY2021S1-CS2103-F10-3/tp/issues/131)

### LINKED ISSUE #167: Exception Handling in UI
**Current**
- No error message thrown

**Expected**
- Should show human friendly error message for all commands if error

**Issues**
[[PE-D] Handling of Max Int · Issue #124 · AY2021S1-CS2103-F10-3/tp](https://github.com/AY2021S1-CS2103-F10-3/tp/issues/124)

### PR DESCRIPTION
**Pull request by therizhao**
Approach

- Use Integer.parseInt instead of regex to get number
- isValidInteger method checks if is within range

Problems fixed

- Remove preceding zeroes for numbers
- Log
    - Rep
        - add, edit → restrict to 1-1000
- Exercise
    - Calories
        - addex, editex → restrict to 1-1000, should not accept numbers

**Link with screenshots of each state**
https://www.notion.so/Fix-boundary-numbers-7eeed0e5911349b98743a79d4d9b1bb9

### CODE REVIEW

**Review by Jillzyt — CHANGES_REQUESTED** [review]
Maybe update UG to indicate the range of 1 to 1000 for reps/calories?

File: src/main/java/seedu/fma/commons/util/IntegerUtil.java [hunk 1]
```diff
@@ -6,20 +6,11 @@
 public class IntegerUtil {
 
     /**
-     * Returns true if a integer more than zero.
-     * @param integer integer to be checked
+     * Returns true iff integer is in range 1-1000 inclusive
```
Comment (Jillzyt) [inline src/main/java/seedu/fma/commons/util/IntegerUtil.java:9]: There's a typo here.

**Comment by therizhao** [comment]
> Maybe update UG to indicate the range of 1 to 1000 for reps/calories?

Will update in another pr

**Comment by therizhao** [comment]
Closes #126

Closes #148

Closes #144

Closes #131

**Review by Jillzyt — APPROVED** [review]

**Pull request merged**